c++/src/kj/async-io.h

```diff
@@ -281,6 +281,11 @@ Tee newTee(Own<AsyncInputStream> input, uint64_t limit = kj::maxValue);
 //
 // It is recommended that you use a more conservative value for `limit` than the default.
 
+Own<AsyncOutputStream> newPromisedStream(Promise<Own<AsyncOutputStream>> promise);
+Own<AsyncIoStream> newPromisedStream(Promise<Own<AsyncIoStream>> promise);
+// Constructs an Async*Stream which waits for a promise to resolve, then forwards all calls to the
+// promised stream.
```

Comment:
Guess there's no immediate need for an AsyncInputStream equivalent.

Reply:
Yeah, I just need PromiseOutputStream and didn't want to make another copy, so I didn't put too much effort into this.